api/lql.go

```diff
@@ -172,75 +65,78 @@ func (q LQLQuery) ValidateRange(allowEmptyTimes bool) (err error) {
 	return nil
 }
 
-type LQLQueryResponse struct {
-	Data    []LQLQuery `json:"data"`
-	Ok      bool       `json:"ok"`
-	Message string     `json:"message"`
-}
-
-// LQLService is a service that interacts with the LQL
+// QueryService is a service that interacts with the Queries
 // endpoints from the Lacework Server
-type LQLService struct {
+type QueryService struct {
 	client *Client
 }
 
-func (svc *LQLService) Create(query string) (
-	response LQLQueryResponse,
+func (svc *QueryService) Create(nq NewQuery) (
+	response QueryResponse,
 	err error,
 ) {
-	lqlQuery := LQLQuery{QueryBlob: query}
-	if err = lqlQuery.Validate(true); err != nil {
-		return
-	}
-
-	err = svc.client.RequestEncoderDecoder("POST", apiLQL, lqlQuery, &response)
+	err = svc.client.RequestEncoderDecoder("POST", apiV2Queries, nq, &response)
 	return
 }
 
-func (svc *LQLService) Update(query string) (
-	response LQLQueryResponse,
+func (svc *QueryService) Update(queryID string, uq UpdateQuery) (
+	response QueryResponse,
 	err error,
 ) {
-	lqlQuery := LQLQuery{QueryBlob: query}
-	if err = lqlQuery.Validate(true); err != nil {
+	if queryID == "" {
+		err = errors.New("query ID must be provided")
 		return
 	}
-
-	err = svc.client.RequestEncoderDecoder("PATCH", apiLQL, lqlQuery, &response)
+	err = svc.client.RequestEncoderDecoder(
+		"PATCH",
+		fmt.Sprintf("%s/%s", apiV2Queries, url.QueryEscape(queryID)),
+		uq,
+		&response,
+	)
 	return
 }
 
-func (svc *LQLService) GetQueries() (LQLQueryResponse, error) {
-	return svc.GetByID("")
+func (svc *QueryService) GetQueries() (
+	response QueriesResponse,
+	err error,
+) {
+	err = svc.client.RequestDecoder("GET", apiV2Queries, nil, &response)
+	return
 }
 
-func (svc *LQLService) GetByID(queryID string) (
-	response LQLQueryResponse,
+func (svc *QueryService) GetByID(queryID string) (
```

Comment:
Have you tested that the GET request here returns 404 when the query does NOT exist? 

I say this since many endpoints don't work as expected, would be good that you check that this does work fine.

Reply:
Confirmed; yes.

```
~/Applications/src/lacework/go-sdk ALLY-561* ⇡
❯ curl --location --request GET 'https://qan.qan.corp.lacework.net/api/v2/Queries/MyExampleQuery' --header 'Authorization: Bearer _ee0d0a838d01a4e77cb3170791810722' -I 
HTTP/1.1 404 Not Found
Server: nginx
Date: Thu, 08 Jul 2021 15:51:12 GMT
Content-Type: application/json
Content-Length: 55
Connection: keep-alive
RateLimit-Remaining: 119
RateLimit-Limit: 120
RateLimit-Reset: 0
Strict-Transport-Security: max-age=31536000; includeSubDomains;; preload
```